client/src/app/site/assignments/modules/assignment-poll/components/assignment-poll-detail/assignment-poll-detail.component.ts

```diff
@@ -116,10 +116,14 @@ export class AssignmentPollDetailComponent extends BasePollDetailComponentDirect
                 if (vote.weight > 0) {
                     const optionContent: ViewUser = option.content_object;
                     if (this.poll.isMethodY) {
-                        if (vote.value === `Y`) {
-                            votes[token].votes.push(optionContent.getFullName());
+                        if (this.poll.max_votes_per_option > 1){ // Show how often the option was selected
+                            votes[token].votes.push(Math.floor(vote.weight).toString() + `x ` + optionContent.getFullName());
                         } else {
-                            votes[token].votes.push(this.voteValueToLabel(vote.value));
+                            if (vote.value === `Y`) {
+                                votes[token].votes.push(optionContent.getFullName());
+                            } else {
+                                votes[token].votes.push(this.voteValueToLabel(vote.value));
+                            }
```

Comment:
I know, that there are three layers before your commit, but separate these two layers in an own function or build an object (a class) from it.